packages/animation-curve-block/src/components/AnimationCurveFlyout.tsx

```diff
@@ -0,0 +1,228 @@
+        if (![x1Validation, x2Validation].includes(Validation.Error)) {
+            onAnimationCurveChange({
+                ...animationCurve,
+                ...animationCurvePatch,
+            });
+        }
+    };
+
+    const updateAnimationFunction = (animationFunctionPatch: AnimationFunctionPatch): void => {
+        updateLocalCurve({
+            ...animationCurve,
+            animationFunction: {
+                ...animationCurve.animationFunction,
+                ...animationFunctionPatch,
+                type: AnimationCurveType.Custom,
+            },
+        });
+    };
+
+    const updateAnimationCurveParameters = (parametersPatch: AnimationCurveParametersPatch): void => {
+        updateAnimationFunction({
+            parameters: {
+                ...animationCurve.animationFunction.parameters,
+                ...parametersPatch,
+            },
+        });
+    };
+
+    return (
+        <Flyout
+            trigger={<></>}
+            isOpen={isFlyoutOpen}
+            onOpenChange={onOpenChange}
+            onCancel={onCancel}
+            legacyFooter={false}
+            fixedFooter={
+                <FlyoutFooter
+                    buttons={[
+                        {
+                            style: ButtonStyle.Default,
+                            emphasis: ButtonEmphasis.Default,
+                            children: 'Cancel',
+                            onClick: onCancel,
+                        },
+                        {
+                            style: ButtonStyle.Default,
+                            emphasis: ButtonEmphasis.Strong,
+                            icon: <IconCheckMark16 />,
+                            children: 'Save',
+                            onClick: handleSave,
+                            disabled: [x1Validation, x2Validation].includes(Validation.Error),
+                        },
+                    ]}
+                />
+            }
+            offset={20}
+            placement={FlyoutPlacement.BottomLeft}
+        >
+            <div className="tw-p-6 tw-w-[498px]">
+                <Dropdown
+                    activeItemId={animationCurve.animationFunction.type}
+                    onChange={(id) =>
+                        onAnimationCurveChange({
+                            ...animationCurve,
+                            animationFunction: {
+                                ...animationCurve.animationFunction,
+                                type: id as AnimationCurveType,
+                                parameters:
+                                    id === AnimationCurveType.Custom
+                                        ? { ...animationCurve.animationFunction.parameters }
+                                        : defaultAnimationCurveTypeValues[id as AnimationCurveType],
+                            },
+                        })
+                    }
+                    menuBlocks={DROPDOWN_MENU_ITEMS}
```

Comment:
Choosing presets from the dropdown does not update the animation curve in the flyout. I think it would be awesome if it did, because right now you can only see the selected curve after clicking close.

Reply:
Fixed that bug